api/v4/common_types.go

```diff
@@ -348,9 +361,38 @@ type AppSourceDefaultSpec struct {
 	// +optional
 	VolName string `json:"volumeName,omitempty"`
 
-	// Scope of the App deployment: cluster, clusterWithPreConfig, local. Scope determines whether the App(s) is/are installed locally or cluster-wide
+	// Scope of the App deployment: cluster, clusterWithPreConfig, local, premiumApps. Scope determines whether the App(s) is/are installed locally, cluster-wide or its a premium app
 	// +optional
 	Scope string `json:"scope,omitempty"`
+
+	// Properties for premium apps, fill in when scope premiumApps is chosen
+	// +optional
+	PremiumAppsProps PremiumAppsProps `json:"premiumAppsProps,omitempty"`
+}
+
+// PremiumAppsProps represents properties for premium apps such as ES
+type PremiumAppsProps struct {
+	// Type: enterpriseSecurity for now, can accomodate itsi etc.. later
+	// +optional
+	Type string `json:"type,omitempty"`
+
+	// Enterpreise Security App defaults
+	// +optional
+	EsDefaults EsDefaults `json:"esDefaults,omitempty"`
+}
```

Comment:
Considering the future needs, can we define a new type called `PremiumAppsType`, then use it in the ES specific data structure?

Reply:
The `type` here represents the type of premium app. It is under `PremiumAppsProps` which represents properties for premiumApps. The `type` field can have `enterpriseSecurity`, `itsi` or other premiumApps as values. 

With the suggestion, wouldn't we be incorrectly putting the type under the purview of the ES app(while it can mean itsi also)? Could you provide a psuedo struct as an example(to understand better)?